src/modules/shopgate/helpers/basket.php

```diff
@@ -204,6 +206,27 @@ protected function loadOrderBasketPayment(shopgate_oxbasket &$oxBasket, Shopgate
     }
 
     /**
+     * @param shopgate_oxbasket $oxidBasket
+     * @param ShopgateCartBase  $shopgateCart
+     *
+     * @post if $oxidBasket supports the setPrice method price will be overwritten
+     */
+    protected function loadOrderBasketTotal(shopgate_oxbasket $oxidBasket, ShopgateCartBase $shopgateCart)
+    {
+        if (!method_exists($oxidBasket,"setPrice")) {
```

Comment:
having a negative condition, just trigger an early return, while not having a return in the other case looks kinda messy to me.
I suggest to have a positive condition and the main code in that if clause.

This leads to a nesting, but this is only a problem the ~ third level